File: src/AdventureWorks.Application/DataEngine/Dbo/AWBuildVersion/Commands/Queries/GetAWBuildVersions/AWBuildVersionLookupModel.cs
using System;
using AutoMapper;
using AdventureWorks.Application.Interfaces.Mapping;
using AdventureWorks.BaseDomain.Entities.Dbo;
using System.Collections.Generic;
using AdventureWorks.Common.Extensions;
using AdventureWorks.BaseDomain.Enums;
using AdventureWorks.BaseDomain.Interfaces;
using AdventureWorks.BaseDomain.CustomTypes;

namespace AdventureWorks.Application.DataEngine.Dbo.AWBuildVersion.Queries.GetAWBuildVersions
{
    public partial class AWBuildVersionLookupModel : IHaveCustomMapping
    {
        public byte SystemInformationID { get; set; }
        public string DatabaseVersion { get; set; }
        public DateTime VersionDate { get; set; }
        public DateTime ModifiedDate { get; set; }

        #region Navigation Properties

        // Uncomment the following line if you need child navigation properties (can cause a poor performance)
        /*  */
        // Uncomment the following line if you need parent navigation properties
        /*  */
        #endregion

        public void CreateMappings(Profile configuration)
        {
            configuration.CreateMap<Domain.Entities.Dbo.AWBuildVersion, AWBuildVersionLookupModel>().IgnoreMissingDestinationMembers().CompensateWithDestinationValues();
            configuration.CreateMap<AWBuildVersionLookupModel, BaseAWBuildVersion>().IgnoreMissingDestinationMembers();
        }
    }
}

File: src/AdventureWorks.Web.Api/ClientApp/src/app/models/data/entities/dbo/a-w-build-version/a-w-build-version-lookup-model.ts
export interface IAWBuildVersionLookupModel {
  systemInformationID: number;
  databaseVersion: string;
  versionDate: Date | string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */
}

export class AWBuildVersionLookupModel implements IAWBuildVersionLookupModel {
  systemInformationID: number;
  databaseVersion: string;
  versionDate: Date | string;
  modifiedDate: Date | string;

  // Uncomment the following line if you need child navigation properties (can cause a poor performance)
  /*  */
  // Uncomment the following line if you need parent navigation properties
  /*  */

  constructor(init?: Partial<AWBuildVersionLookupModel>) {
    if (!!init) {
      Object.assign<AWBuildVersionLookupModel, Partial<AWBuildVersionLookupModel>>(this, init);
    }
  }
}
